code/modules/mob/living/simple_animal/abnormality/zayin/we_can_change_anything.dm

```diff
@@ -25,11 +34,88 @@
 	gift_type =  /datum/ego_gifts/change
 	gift_message = "Your heart beats with new vigor."
 
+	var/grinding = FALSE
+	var/grind_duration = 5 SECONDS
+	var/grind_damage = 2 // Dealt 100 times
+
 /mob/living/simple_animal/hostile/abnormality/we_can_change_anything/work_complete(mob/living/carbon/human/user, work_type, pe)
 	user.apply_damage(40, RED_DAMAGE, null, user.run_armor_check(null, RED_DAMAGE)) // say goodbye to your kneecaps chucklenuts!
 	user.apply_status_effect(STATUS_EFFECT_CHANGE)
 	..()
 
+/mob/living/simple_animal/hostile/abnormality/we_can_change_anything/Move()
+	if(charge_state)
+		return ..()
+	return FALSE
+
+/mob/living/simple_animal/hostile/abnormality/we_can_change_anything/AttackingTarget()
+	if(COOLDOWN_FINISHED(src, charge_cooldown))
+		COOLDOWN_START(src, charge_cooldown, charge_frequency)
+		Grind()
+		return TRUE
+	return FALSE
+
+/mob/living/simple_animal/hostile/abnormality/we_can_change_anything/handle_charge_target(atom/target)
+	if(..())
+		update_all()
+
+/mob/living/simple_animal/hostile/abnormality/we_can_change_anything/charge_end()
+	..()
+	Grind()
+
+/mob/living/simple_animal/hostile/abnormality/we_can_change_anything/proc/Grind()
```

Comment:
You know, it would be cool if it charged in some direction and when it hit SOMEONE, it'd put them in the machine and then deal red damage over time.